enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -126,12 +126,32 @@ type CacheSnapshotOptions struct {
 	WorkspaceFSPath string
 
 	// Labeled map of chunked artifacts backed by blockio.COWStore storage.
-	ChunkedFiles map[string]*blockio.COWStore
+	ChunkedFiles map[string]*ChunkedFile
 }
 
 type UnpackedSnapshot struct {
 	// ChunkedFiles holds any chunked files that were part of the snapshot.
-	ChunkedFiles map[string]*blockio.COWStore
+	ChunkedFiles map[string]*ChunkedFile
+}
+
+type ChunkedFile struct {
+	*blockio.COWStore
+
+	// digests caches any known digests for *non-dirty* chunks, keyed by offset.
+	digests map[int64]string
```

Comment:
Why not just cache the full digest including size? So we don't always have to recompute it